wasm.md

```diff
@@ -1448,6 +1437,58 @@ Start Function
          </moduleInst>
 ```
 
+Identifiers
+-----------
+
+Though each part of a module (each function, global, etc.) primarily has an index, the text format also allows us to refer to them by identifiers.
+We commonly want to store these optional identifiers when allocating something.
+In KWasm we store identifiers in maps from `Identifier` to `Int`, the `Int` being an index.
+This rule handles storing an `OptionalId` in the correct lookup table.
+
+```k
+    syntax Instr ::= "saveId" "(" String "," OptionalId "," Int ")"
```

Comment:
Could this be made cleaner somehow? I don't like all the code duplication, but it is nice to get it out of the regular semantics rules.

Reply:
Well first you can do `syntax Instr ::= saveId ( String , OptionalId , Int )` (without the quotes).

Second, I think in every case (I scanned briefly) it looks like we can just inline the `saveId` into the place where it's used.

Reply:
I guess you have a separate rule so that you can handle the case `notBool isIdentifier(...)` separately, but will anything go wrong if you keep over-writing `IDMAP [ .Identifier <- SOME_INDEX ]`? It will never be accessed.

Another option is to make a function `syntax Map ::= addIdentifier ( Map , Identifier , Int )`, which _functionally_ implements this same logic. That way you can just say `<typeIds> IDS => addIdentifier(IDS, OID, IDX) </typeIds>` _inline_ in the places where `saveId` is used. `addIdentifier` will just return the original `Map` if it's `notBool isIdentifier`, otherwise will actually do the map update.

Reply:
Done